### PR TITLE
feat: Private app OIDC authentication

### PR DESCRIPTION
**Pull request by jachym-tousek-keboola**
Jira: https://keboola.atlassian.net/browse/PSGO-364

**Changes:**
- Private app OIDC authentication

-----------

### CODE REVIEW

**Review by michaljurecko — COMMENTED** [review]
LGTM

Please add assertions to the response body that have some error message for the user, see comments.

File: internal/pkg/service/appproxy/http/router_test.go [hunk 1]
```diff
@@ -1,56 +1,422 @@
...
+		},
+		{
+			name: "public-app-sub-url",
+			run: func(t *testing.T, handler http.Handler, m *mockoidc.MockOIDC) {
+				// Request to public app
+				rec := httptest.NewRecorder()
+				req := httptest.NewRequest("GET", "https://public.data-apps.keboola.local/some/data/app/url?foo=bar", nil)
+				handler.ServeHTTP(rec, req)
+				assert.Equal(t, http.StatusOK, rec.Code)
+				assert.Equal(t, "Hello, client", rec.Body.String())
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/appproxy/http/router_test.go:74]: LGTM, in the next work, we should add asserts for `X-Forwarded` headers.

File: internal/pkg/service/appproxy/http/router_test.go [hunk 2]
```diff
@@ -1,56 +1,422 @@
...
+					req.Header.Add("Cookie", cookie)
+				}
+				handler.ServeHTTP(rec, req)
+				require.Equal(t, http.StatusOK, rec.Code)
+			},
+		},
+		{
+			name: "private-app-unauthorized-valid-csrf",
+			run: func(t *testing.T, handler http.Handler, m *mockoidc.MockOIDC) {
+				m.QueueUser(&mockoidcCustom.MockUser{
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/appproxy/http/router_test.go]: :heavy_check_mark:

**Review by michaljurecko — APPROVED** [review]
LGTM

File: internal/pkg/service/appproxy/http/app.go [hunk 3]
```diff
@@ -41,3 +43,62 @@ func publicAppHandler(app DataApp, _ config.Config, chain alice.Chain) (http.Han
...
+
+func protectedAppHandler(app DataApp, cfg config.Config, chain alice.Chain) (http.Handler, error) {
+	authValidator := func(email string) bool {
+		// No need to verify users, just groups which is done using AllowedGroups in provider configuration.
+		return true
+	}
+
+	config, err := authProxyConfig(app, cfg, chain)
+	if err != nil {
+		return nil, err
```
Comment (hosekpeter) [inline internal/pkg/service/appproxy/http/app.go:55]: Why don't we return a specific error.  fmt.Errorf("some text")

**Review by hosekpeter — APPROVED** [review]
LGTM

**Pull request merged**